packages/flutter_tools/test/general.shard/flutter_platform_test.dart

```diff
@@ -57,6 +59,50 @@ void main() {
       ProcessManager: () => FakeProcessManager.any(),
     });
 
+    group('Observatory and DDS setup', () {
+      MockPlatform mockPlatform;
+      MockProcessManager mockProcessManager;
+      FlutterPlatform flutterPlatform;
+      final Map<Type, Generator> contextOverrides = <Type, Generator>{
+        Platform: () => mockPlatform,
+        ProcessManager: () => mockProcessManager,
+        FileSystem: () => fileSystem,
+      };
+
+      setUp(() {
+        mockPlatform = MockPlatform();
+        when(mockPlatform.isWindows).thenReturn(false);
+        mockProcessManager = MockProcessManager();
+        flutterPlatform = TestObservatoryFlutterPlatform();
+      });
+
+      Future<List<String>> captureCommand() async {
```

Comment:
Consider using the FakeProcessManager and listing out the commands explicitly, like:

https://github.com/flutter/flutter/blob/943b41bd61de434fd18777675b89ade479ccefd9/packages/flutter_tools/test/general.shard/macos/macos_device_test.dart#L61

Reply:
Yes! this is much easier. Do you have any concern that is more brittle, since any changes to other flags of the command would break it?

Reply:
We want the tests to fail if the commands/flags change